### PR TITLE
Fix incorrect signature test vectors in BIP322

### PR DESCRIPTION
**Pull request by wip-abramson**
While attempting to implement bip322 I was unable to generate the same signature as the ones provided. Furthermore, I was unable to verify these signatures in my implementation. After some digging, I noticed the test vectors in the bip322 PR to Bitcoin Core uses different values.

See util_tests.cpp in https://github.com/bitcoin/bitcoin/pull/24058

This P.R. replaces the current test vectors with the ones from the P.R.

It also adds additional test vectors for the transaction hashes of the to_spend and to_sign transactions. Having these would have speeded up my debugging process significantly.

### CODE REVIEW

**Comment by rxgrant** [comment]
Can you point at the code needed to generate these vectors?

Can you test setting version to 2 to see if the old vectors were leftover cruft from before [this fix](/bitcoin/bitcoin/pull/24058#issuecomment-1027547066)?

**Comment by wip-abramson** [comment]
Setting the version to 2 still did not produce the test vectors currently provided in this bip.

I am not sure how these test vectors were generated, both the ones in the bip and the ones used in the P.R. - perhaps a question for @kallewoof

File: bip-0322.mediawiki [hunk 1]
```diff
@@ -174,7 +174,20 @@ Given below parameters:
 * private key <code>L3VFeEujGtevx9w18HD1fhRbCH67Az2dpCymeRE1SoPK6XQtaN2k</code>
 * corresponding address <code>bc1q9vza2e8x573nczrlzms0wvx3gsqjx7vavgkx0l</code>
 
+
```
Comment (kallewoof) [inline bip-0322.mediawiki:177]: nit: Unnecessary whitespace.

File: bip-0322.mediawiki [hunk 2]
```diff
@@ -174,7 +174,20 @@ Given below parameters:
...
 Produce signatures:
 
-* Message = "" (empty string): <code>AkcwRAIgFuS8y5m0ym9Gj2odoVB5NIL+cPYkeEj8LL1N/6P58X8CIA6jJ9QH2iYKRXVfmhsDzHq1bMS4Adj0nb8DDSdN/SpBASECx/EgAxlkQpQ9hYjgGu6EBCPMVPwVIVJqO4XCsMvViHI=</code>
-* Message = "Hello World": <code>AkcwRAIgG3PASL/vRTgAqogWT6S8rUOQXNnfRzX6JncmbFlHc1ACIGQdsW+rnVmsQzyAYRQisHKFMigDmKiL7LUw4x17Fw5tASECx/EgAxlkQpQ9hYjgGu6EBCPMVPwVIVJqO4XCsMvViHI=</code>
+* Message = "" (empty string): <code>AkcwRAIgM2gBAQqvZX15ZiysmKmQpDrG83avLIT492QBzLnQIxYCIBaTpOaD20qRlEylyxFSeEA2ba9YOixpX8z46TSDtS40ASECx/EgAxlkQpQ9hYjgGu6EBCPMVPwVIVJqO4XCsMvViHI=</code>
+* Message = "Hello World": <code>AkcwRAIgZRfIY3p7/DoVTty6YZbWS71bc5Vct9p9Fia83eRmw2QCICK/ENGfwLtptFluMGs2KsqoNSk89pO7F29zJLUx9a/sASECx/EgAxlkQpQ9hYjgGu6EBCPMVPwVIVJqO4XCsMvViHI=</code>
```
Comment (kallewoof) [inline bip-0322.mediawiki:181]: Verified.

```
$ ./bitcoin-cli signmessage bc1q9vza2e8x573nczrlzms0wvx3gsqjx7vavgkx0l ""
AkcwRAIgM2gBAQqvZX15ZiysmKmQpDrG83avLIT492QBzLnQIxYCIBaTpOaD20qRlEylyxFSeEA2ba9YOixpX8z46TSDtS40ASECx/EgAxlkQpQ9hYjgGu6EBCPMVPwVIVJqO4XCsMvViHI=
$ ./bitcoin-cli signmessage bc1q9vza2e8x573nczrlzms0wvx3gsqjx7vavgkx0l "Hello World"
AkcwRAIgZRfIY3p7/DoVTty6YZbWS71bc5Vct9p9Fia83eRmw2QCICK/ENGfwLtptFluMGs2KsqoNSk89pO7F29zJLUx9a/sASECx/EgAxlkQpQ9hYjgGu6EBCPMVPwVIVJqO4XCsMvViHI=
$
```

**Comment by kallewoof** [comment]
Nit: typo in title of pull request ("fixe").

**Comment by kallewoof** [comment]
> I am not sure how these test vectors were generated, both the ones in the bip and the ones used in the P.R. - perhaps a question for @kallewoof

Forgot to reply: I believe the test vectors in the BIP here were based on an outdated version. @rxgrant let me know if you have issues or a different opinion on the matter.

File: bip-0322.mediawiki (hunk 1 above)
Comment (wip-abramson) [inline bip-0322.mediawiki:177]:
```suggestion
```

**Comment by kallewoof** [comment]
Squash?

**Comment by wip-abramson** [comment]
Sorry @kallewoof are you asking if we should squash, or asking me to squash?

Happy to squash if that is preferable.

**Comment by kallewoof** [comment]
↑

**Comment by wip-abramson** [comment]
Is this squashed? I seem to just be adding more commits.

Apologies, I don't think I have ever attempted to squash before

**Comment by michaelfolkson** [comment]
wip-abramson: I don't think this BIPs repo is as strict on that as the Core repo is but one of the BIP editors can correct me if wrong. It is definitely better practice though for any future PRs made to this repo.

**Comment by kallewoof** [comment]
Typically not as strict, but no need to be overly committy ;)
